applications/RomApplication/python_scripts/hrom_training_utility.py

```diff
@@ -192,19 +178,60 @@ def __AppendHRomWeightsToRomParameters(self):
             # Add the missing parents to the elements dict with a null weight
             for parent_id in missing_condition_parents:
                 hrom_weights["Elements"][parent_id] = 0.0
+            w, z = self.__AddSelectedElementsWithZeroWeights(w,z, missing_condition_parents)
 
-        # Append weights to RomParameters.json
-        # We first parse the current RomParameters.json to then append and edit the data
-        with open('RomParameters.json','r') as f:
-            updated_rom_parameters = json.load(f)
-            #FIXME: I don't really like to automatically change things without the user realizing...
-            #FIXME: However, this leaves the settings ready for the HROM postprocess... something that is cool
-            #FIXME: Decide about this
-            # updated_rom_parameters["train_hrom"] = False
-            # updated_rom_parameters["run_hrom"] = True
-            updated_rom_parameters["elements_and_weights"] = hrom_weights #TODO: Rename elements_and_weights to hrom_weights
 
-        with open('RomParameters.json','w') as f:
-            json.dump(updated_rom_parameters, f, indent = 4)
+        np.save('WeightsMatrix.npy',w)
+        np.save('ElementsVector.npy',z)
 
         if self.echo_level > 0 : KratosMultiphysics.Logger.PrintInfo("HRomTrainingUtility","\'RomParameters.json\' file updated with HROM weights.")
+
+
+
+    def __AddSelectedElementsWithZeroWeights(self, original_weights,original_elements, elements_to_add):
+
+        added_elements_ids = np.array(elements_to_add)
+        updated_elements = np.r_[original_elements, added_elements_ids]
+        updated_weights = np.r_[original_weights, np.zeros(added_elements_ids.shape)]
+
+        return updated_weights, updated_elements
+
+
+
+    def __AddSelectedConditionsWithZeroWeights(self, original_weights, original_conditions, conditions_to_add, number_of_elements):
+
+        added_conditions_ids = np.array(conditions_to_add)+number_of_elements
+        updated_conditions = np.r_[original_conditions, added_conditions_ids]
+        updated_weights = np.r_[original_weights, np.zeros(added_conditions_ids.shape)]
+
+        return updated_weights, updated_conditions
+
+
+    def __CreateDictionaryWithRomElementsAndWeights(self, w = None, z=None, n_elements = None):
```

Comment:
does it make sense to still call this "CreateDictionary"?

Reply:
I think it does, it returns a dictionary.